internal/lookup_compound.go

```diff
@@ -22,20 +22,17 @@ func parseWords(phrase string, preserveCase bool, splitBySpace bool) []string {
 	}
 
 	// Regex pattern to match words, including handling apostrophes
-	var pattern string
-	if preserveCase {
-		pattern = `([\p{L}\d]+(?:['’][\p{L}\d]+)?)`
-	} else {
+	pattern := `([\p{L}\d]+(?:['’][\p{L}\d]+)?)`
+	if !preserveCase {
 		phrase = strings.ToLower(phrase)
-		pattern = `([\p{L}\d]+(?:['’][\p{L}\d]+)?)`
 	}
 
 	re := regexp.MustCompile(pattern)
 	return re.FindAllString(phrase, -1)
 }
 
 func (s *SymSpell) LookupCompound(phrase string, maxEditDistance int) *items.SuggestItem {
```

Comment:
Parsing regexp has a cost. For this reason, regex should be computed only once, and not eaxh time a method is called.

Also, you are MustCompile can cause a panic. Here it would be at runtime if you ever update the regexp sometimes later.

For this two reasons, it's better to do it at compilation time.

```suggestion
 	return reSplit.FindAllString(phrase, -1)
 }

var reSplit = regexp.MustCompile(`([\p{L}\d]+(?:['’][\p{L}\d]+)?)`)
 
 func (s *SymSpell) LookupCompound(phrase string, maxEditDistance int) *items.SuggestItem {
```